## The landscape of gain and loss of function mutations in melanoma
![f1](https://github.com/user-attachments/assets/cfc1621f-ed66-4fb3-8901-0751907b4595)

#### This repository contains the code used to perform analyses and create figures as described in the manuscript [1]. 


The code is structured as follows:

**omics_models**

Contains scripts used for employing AlphaMissense to classify GOF/LOF variants by structural context. 

**pathway_analysis**

Contains scripts used to calculate pathway scores for each omics type (whole-exome sequencing, RNA-sequencing, RPPA), assemble correlation matrices for each phenotype of interest and treatment type, and plot the results as corrplots.

**variant_classification**

Contains scripts for generating omics specific factor loadings, annotating variants with GOF/LOF label, classification, and downstream analyses/visualizations.


[1] <i> Konstantinos Tsingas*, Monzy Thomas*, Bryant Duong*, Min Xiao*, Jayamanna Wickramasinghe, Yeqing Chen, McKenna Reale, Gatha Thacker, Haiyin Li, Haiwei Mou, Eric Ramirez-Salazar, Jessie Villanueva, Xiaowei Xu, Ahron Flowers, Giorgos C. Karakousis, John T. Miura, Tara C. Mitchell, Ravi K. Amaravadi, Lynn M. Schuchter, Shujing Liu, Qi Long, Dave S. B. Hoon, Romela I. Ramos, Matias A. Bustos, Jeffrey E. Gershenwald, Julie G. Simon, Jennifer A. Wargo, Michael A. Davies, Yiling Lu, Gordon B. Mills, Sonia Cohen, Aleigha Lawless, Tatyana Sharova, Dennie T. Frederick, Keith T. Flaherty, Nir Hacohen, Genevieve M. Boland, Noam Auslander#, Meenhard Herlyn#. </i> <b> The landscape of gain and loss of function mutations in melanoma </b>
